content/post/formatting-with-style.md

```diff
@@ -0,0 +1,267 @@
+if it is available. For this reason we're working on the second
+iteration of our format analyzer, which works in two distinct steps:
+first it learns to model style on many repositories, then it is
+applied to a given codebase.
+
+To be clear though, our goal is still __not__ to model _global style
+practices_. For this reason, we don't want to use the standard
+supervised learning approaches but investigate meta-learning
+instead. Meta-learning can be described as the set of techniques that
+allow _learning how to learn_.
+
+The plan is as follows:
+
+1. Learn how to learn to model style in a given repository by
+   leveraging all the repositories which are relevant
+2. Apply this knowledge to model a particular repository
+
+This scheme is quite different from supervised learning.
+
+To achieve the meta learning objective, we consider using well-known
+deep learning techniques:
+
+### Step 1, learning to learn
+
+The first phase and also the most time-consuming one is to train an
+efficient, general model that can learn how to model style in a repo
+with minimal effort.
+
+Therefore we exploit the idea of an _embedding_ to model style, with a
+twist: the embedding is the only place where the model can store
+information about the style of a repository and hence __has__ to
+condense __all the style information__.
+
+### Step 2, applying the formatter to new repositories
+
+Since the style embedding is the only place where our model can encode
+style information about a repository, we can use techniques that are
+well-known in both Natural Language Processing and Computer Vision
+communities to find the embedding that most closely models the style.
+
+And that's all: since all of the style information belongs to one
+single place, we've nothing else to do.
+
+### Why does it work?
+
+The described two-step process is efficient because the adopted model
+architecture can be arbitrarily complex and expressive to produce
+style hints. It can learn how certain styles are applied in certain
+corner cases in millions of repositories.
+
+Yet, it is easy to model style for a single repository since the core
+of what varies is encoded in a central place.
+
+### Possibilities
+
+Meta learning enables lots of new applications. Among the most
+exciting ones we could cite:
+
+- apply a specific style to your repository (Google's, another team's, etc)
+- find existing styles with documented guidelines that most closely match an
+  unknown style
+- perform style analysis at a fine granularity (directory granularity
+  instead of project granularity). This is made possible by the
+  reduced need for data during the second step
+
+## To be continued!
+
+We are making early steps towards the second approach. Let's wish
+Format Analyzer a bright future that will allow engineers across the
+world to focus on problems more exciting than maintaining the proper
+code formatting.
+
+Don't want to miss the next blog post about how source{d} ML team does
```

Comment:
I think these CTA should be secondary, the main CTA is trying Lookout on a test repo.